tiny_dnn/util/math_functions.h

```diff
@@ -34,59 +34,67 @@ namespace tiny_dnn {
 
 // x = x / denom
 inline void vector_div(vec_t& x, float_t denom) {
-    std::transform(x.begin(), x.end(), x.begin(), [=](float_t x) { return x / denom; });
+    std::transform(x.begin(), x.end(), x.begin(),
+                   [=](float_t x) { return x / denom; });
 }
 
 namespace detail {
-    inline void moments_impl_calc_mean(serial_size_t num_examples, serial_size_t channels,
-                                       serial_size_t spatial_dim, const tensor_t& in, vec_t& mean) {
-        for (serial_size_t i = 0; i < num_examples; i++) {
-            for (serial_size_t j = 0; j < channels; j++) {
-                float_t& rmean = mean.at(j);
-                const auto it = in[i].begin() + (j * spatial_dim);
-                rmean = std::accumulate(it, it + spatial_dim, rmean);
-            }
+inline void moments_impl_calc_mean(size_t num_examples,
+                                   size_t channels,
+                                   size_t spatial_dim,
+                                   const tensor_t& in, vec_t& mean) {
+    for (size_t i = 0; i < num_examples; i++) {
+        for (size_t j = 0; j < channels; j++) {
+            float_t& rmean = mean.at(j);
+            const auto it = in[i].begin() + (j * spatial_dim);
```

Comment:
@yumetodo for me the word 'should' means that it might not happen and we want this as much optimized as possible. But don't worry we can fix this later.